### ISSUE TITLE
[Feature Request] The email sign in support verification code.

### ISSUE THREAD

**Issue by ahaapple** [memfreeme/memfree#68]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

Currently, the way to log in via email is to send a magic link, which is not very user-friendly. So we hope to support the way to also send verification codes via email.

We can allow users to click on the link, or we can ask them to enter a verification code (when on different devices)

The verification code can be generated based on the token, 6 digits

The related code is here:
frontend/lib/auth/auth-sind-request.ts

### Use Case

Make user email verification more friendly.

### Priority

Medium

### Record

- [X] I have read the Contributing Guidelines
- [X] I have starred the repository

**Comment by Che140401** [comment]
Hi @ahaapple,

I have 2 years of experience as a web developer, and I would love to work on implementing the email verification code feature. Additionally, I can focus on making the verification process more user-friendly. Could you please assign this issue to me?

Thanks!

**Comment by ahaapple** [comment]
@Che140401 Assigned, Thank you!

**Comment by rohitmore07** [comment]
Hey @ahaapple I am very keen to work on this issue. I would be really happy if you could assign this issue to me.

**Comment by rohitmore07** [comment]
@ahaapple Got it! To clarify, I need to add a new login page with :

**An email login section supporting both magic links and verification codes.
A "Sign in with Google" button below the email login form.**

Please confirm if this is correct, and I'll proceed.

**Comment by ahaapple** [comment]
Yes. we could let the user input verification codes with  https://ui.shadcn.com/docs/components/input-otp.

**Comment by rohitmore07** [comment]
Okay

**Comment by rohitmore07** [comment]
I encountered an issue while working with your repository.
Could you please help me resolve this? Let me know if you need any additional details.
Thanks in advance!

![Image](https://github.com/user-attachments/assets/267b8b29-bc74-481b-bc1d-3bddfc0ab65e)
